### PR TITLE
Issue 2: Save Image Link

### PR DESCRIPTION
**Pull request by kkirasich**

### CODE REVIEW

**Comment by kkirasich** [comment]
[Issue 2: Save Image Link](https://trello.com/c/jONlsavi/5-issue-2-save-image-link)

File: test/models/image_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class ImageTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (darreneng) [inline test/models/image_test.rb]: here we're going to need some tests:

- valid URL case (valid image)
- some string that isn't valid
- empty string

File: db/migrate/20190319195942_create_images.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateImages < ActiveRecord::Migration[5.2]
+  def change
+    create_table :images do |t|
+      t.string :url
```
Comment (darreneng) [inline db/migrate/20190319195942_create_images.rb]: might want to add `null: false` here. I'm not sure if columns are non nullable by default

File: app/views/images/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<h1>Enter URL</h1>
+<%= simple_form_for @image do |form| %>
+  <p>
+    <%= form.label :url %><br>
```
Comment (darreneng) [inline app/views/images/new.html.erb]: we can try to use simple form here and have simple form display the errors for us. https://github.com/plataformatec/simple_form#usage

File: app/views/images/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Image Uploaded</h1>
+<img src="<%= @image.url %>" />
```
Comment (darreneng) [inline app/views/images/show.html.erb]: check out https://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    assert_redirected_to image_path(Image.last.id)
+  end
+
+  def test_create_fail
+    assert_difference('Image.count', 0) do
+      post images_url, params: { image: { url: 'BAD_URL_TEST' } }
+    end
+
+    assert_select '#url-error', 'Url must be a valid URL'
```
Comment (darreneng) [inline test/controllers/images_controller_test.rb]: test the response code here

File: app/controllers/images_controller.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    @image = Image.new
+  end
+
+  def create
+    @image = Image.new(params[:image].permit(:url))
+
+    if @image.save
+      redirect_to @image
+    else
+      render :new
```
Comment (darreneng) [inline app/controllers/images_controller.rb]: probably want a different response code for this path

**Pull request merged**